src/drivers/sifive,timer.c

```diff
@@ -0,0 +1,45 @@
+/* Copyright 2018 SiFive, Inc */
+/* SPDX-License-Identifier: Apache-2.0 */
+
+#include <mee/drivers/sifive,timer.h>
+
+int __mee_driver_sifive_timer_get(struct mee_timer *timer, unsigned long *tv)
+{
+    struct __mee_driver_timer *mee_timer = (void *)(timer);
+    unsigned long val = 0;
+
+    #if __riscv_xlen == 32
+        unsigned int hi, hi1, lo;
+
+        asm volatile ("csrr %0, mcycleh" : "=r"(hi));
+        asm volatile ("csrr %0, mcycle" : "=r"(lo));
+        asm volatile ("csrr %0, mcycleh" : "=r"(hi1));
```

Comment:
Ah, if the timer is from `mcycle` then this driver should really be a driver for a hart.  Since the timebase could differ between harts, then the interface should contain a way to obtain a timer structure that corresponds to each hart.